util/dockershell/compose/compose.go

```diff
@@ -91,7 +91,118 @@ func WithStdio(stdout, stderr io.Writer) Option {
 	}
 }
 
-// New creates a new Compose of the specified docker-compose yaml data.
+// Builds all services defined in the docker-compose yaml
+
+func Build(dockerComposeYaml string, opts ...Option) ([]func() error, error) {
+	var cOpts options
+	for _, o := range opts {
+		o(&cOpts)
+	}
+	tmpContext, err := os.MkdirTemp("", "compose"+xid.New().String())
+	if err != nil {
+		return nil, err
+	}
+	confFile := filepath.Join(tmpContext, "docker-compose.yml")
+	if err := os.WriteFile(confFile, []byte(dockerComposeYaml), 0600); err != nil {
+		return nil, err
+	}
+
+	var cleanups []func() error
+	for i := 0; i < 3; i++ {
```

Comment:
The first and second calls to compose down with `--rmi` flag set do not remove the images but rather it untags them. Only on the third call are images removed.

Reply:
curious why is this 🤔, since the command already has `--rmi all` and why both 1st and 2nd only untag them.

Reply:
The `all` flag will remove all images regardless if they have a custom tag or not. The `local` flag will remove everything that doesn't have a custom tag. 

> why both 1st and 2nd only untag them

Yeah I'm not sure why that is, maybe as a safety net or just to keep images for cache in case you wanted to restart services.